core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -99,7 +99,11 @@ public UnsafeInMemorySorter(
     this.consumer = consumer;
     this.memoryManager = memoryManager;
     this.sorter = new Sorter<>(UnsafeSortDataFormat.INSTANCE);
-    this.sortComparator = new SortComparator(recordComparator, prefixComparator, memoryManager);
+    if (recordComparator != null) {
```

Comment:
`recordComparator` is nullable? That's surprising to me. If `recordComparator` is null, does that also imply that `prefixComparator` is null? Can we put `@Nullable` annotations on the parameter so that IntelliJ's fancy nullability handling checks work properly?